src/Jackett.Common/Indexers/FileList.cs

```diff
@@ -99,109 +100,169 @@ public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken
             });
             return IndexerConfigurationStatus.RequiresTesting;
         }
-
-        protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query)
+        private NameValueCollection getQueryCollection(String imdbID, String searchString, String cat, int page)
         {
-            var releases = new List<ReleaseInfo>();
-            var searchUrl = BrowseUrl;
-            var searchString = query.GetQueryString();
-            var cat = MapTorznabCapsToTrackers(query).FirstIfSingleOrDefault("0");
-
             var queryCollection = new NameValueCollection();
 
-            if (query.ImdbID != null)
-                queryCollection.Add("search", query.ImdbID);
+            if (imdbID != null)
+            {
+                queryCollection.Add("search", imdbID);
+            }
             else if (!string.IsNullOrWhiteSpace(searchString))
+            {
                 queryCollection.Add("search", searchString);
+            }
 
             queryCollection.Add("cat", cat);
             queryCollection.Add("searchin", "1");
             queryCollection.Add("sort", "2");
+            queryCollection.Add("page", page.ToString());
 
-            searchUrl += "?" + queryCollection.GetQueryString();
+            return queryCollection;
+        }
 
-            var response = await RequestStringWithCookiesAndRetry(searchUrl, null, BrowseUrl);
+        private async Task<string> callProviderAsync(TorznabQuery query, int page)
+        {
+            var searchUrl = BrowseUrl;
+            var searchString = query.GetQueryString();
+
+            var cat = MapTorznabCapsToTrackers(query).FirstIfSingleOrDefault("0");
+
+            searchUrl += "?" + getQueryCollection(query.ImdbID, searchString, cat, page).GetQueryString();
 
+            var response = await RequestStringWithCookiesAndRetry(searchUrl, null, BrowseUrl);
             // Occasionally the cookies become invalid, login again if that happens
             if (response.IsRedirect)
             {
                 await ApplyConfiguration(null);
                 response = await RequestStringWithCookiesAndRetry(searchUrl, null, BrowseUrl);
             }
 
-            var results = response.Content;
+            return response.Content;
+        }
+
+        protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query)
+        {
+            var releases = new List<ReleaseInfo>();
+            var pageCount = 0;
+            var firstResponse = await callProviderAsync(query, 0);
+
+            pageCount = getPageCount(firstResponse);
+            transformProviderRequest(query, firstResponse, releases);
+            //Wait
+            Thread.Sleep(50);
+
+            //Process the rest of the pages
+            if (pageCount > 0 && query.GetQueryString() != "")
```

Comment:
Is this code fetching every page from the results of a search?
If that is the case, then consider that this approach can be misconstrued by the FileList server as an DDoS attack, as we will be flooding the server with multiple http GET requests in a very short period of time. The Jackett user's IP may get blacklisted.
The Torznab default is to return about 100 rows. So if a FileList page has 20 rows then we should aim to fetch no more than 5 pages.
Also, I suggest spreading out the gap between requests so as not to flood the server with too many at once. I recommend a pause of 250ms between requests rather than the current 50ms wait.
Thanks.

Reply:
@garfield69 As I suggested here: https://github.com/Jackett/Jackett/issues/5190#issuecomment-605620438 I really do think it's best to discuss this with the FileList staff. They now have an API that needs to be implemented #7004 and also #5190 may be solved in a different way if we talk to them first. I am happy to start a conversation with them if you guys are fine with that.

As much as I like the outcome of this change, I don't want the IPs and accounts of every FL Jackett user, myself included, to get banned because of suspected DDoS attacks.

Thoughts? Need input from everyone here:
@ngosang @cadatoiva @IIIspaceIII

Reply:
We could limit to 5 pages I think.
Also we could increase the time between requests to 250ms.